unilog_test.go

```diff
@@ -83,3 +85,65 @@ func TestFilterFunction(t *testing.T) {
 		t.Errorf("expected %q, found %q", expected, result)
 	}
 }
+
+func TestTwoStateExit(t *testing.T) {
+	r := strings.NewReader(strings.Join(shakespeare, "\n"))
+	u := &Unilog{}
+	exitCode := -1
+
+	term := make(chan os.Signal, 2)
+	quit := make(chan os.Signal, 2)
+	exit := make(chan int, 2)
+
+	u.sigTerm = term
+	u.sigQuit = quit
+	u.exit = func(code int) {
+		exitCode = code
+		exit <- 1
+	}
+	u.shutdown = make(chan struct{})
+
+	go u.run()
+
+	term <- syscall.SIGTERM
+
+	u.lines, u.errs = readlines(r, u.BufferLines, u.shutdown)
+
+	term <- syscall.SIGTERM
+	<-u.shutdown
+	quit <- syscall.SIGQUIT
+
+	<-exit
+	if exitCode != 1 {
+		t.Error("Did not call exit.")
+	}
+}
+
+func TestNoExitOnQuit(t *testing.T) {
+	r := strings.NewReader(strings.Join(shakespeare, "\n"))
+	u := &Unilog{}
+	exitCode := -1
+
+	term := make(chan os.Signal, 2)
+	quit := make(chan os.Signal, 2)
+
+	u.sigTerm = term
+	u.sigQuit = quit
+	u.exit = func(code int) {
+		exitCode = code
+	}
+	u.shutdown = make(chan struct{})
+
+	go u.run()
+
+	term <- syscall.SIGTERM
```

Comment:
Hrm, I think I understand why you have to send to the sigterm chan but it feels like this is making this test a bit more unclear. Wonder if it would make sense to include an (otherwise unused) `stopRunning` chan?